File: src/org/genyris/test/java/JavaDummy.java
package org.genyris.test.java;


public class JavaDummy {
	//
	// Class for testing the FFI code with.
	//
	public static int staticField = 123;
	public int intField;
	public long longField;
	public char charField;
	public float floatField;
	public double doubleField;
	public boolean booleanField;
	public int byteField;
	public int shortField;
	public String stringField;

	public int privateField;

	public JavaDummy() {
		byteField = 0xFF;
		shortField = 2;
		intField = 42;
		longField = 42000000;
		floatField = (float) 4.2;
		doubleField = 4.2e42;
		charField = 'G';
		booleanField = false;
		stringField = "FOO!";
		
		privateField = 42;
		privateField = privateField*2; // stop compiler warning
	}
	
	public int getInt() {
		return intField;
	}
	public void setInt(int i) {
		intField = i;
	}
	public static String[] staticMethod1(int count) {
		String[] retval = new String[count];
		for(int i=0;i<count;i++) {
			retval[i] = Integer.toString(i);
		}
		return retval;
	}
	public static Integer[] staticMethod2(int count) {
		Integer[] retval = new Integer[count];
		for(int i=0;i<count;i++) {
			retval[i] = new Integer(i);
		}
		return retval;
	}

	public String[] method1(String[] args) {
		return args;
	}
	public static int failmethod1() throws Exception {
		int x = 45;
		if(x == 45) throw new Exception("death in failmethod1()");
		return 43; 
	}
	public int failmethod2() throws Exception {
		int x = 45;
		if(x == 45) throw new Exception("death in failmethod2()");
		return 43; 
	}
}

File: src/org/genyris/test/java/JavaDummy.kt
package org.genyris.test.java


class JavaDummy {
    var int: Int = 42
    var longField: Long = 42000000
    var charField: Char = 'G'
    var floatField: Float
    var doubleField: Double = 4.2e42
    var booleanField: Boolean = false
    var byteField: Int = 0xFF
    var shortField: Int = 2
    var stringField: String? = "FOO!"

    var privateField: Int = 42

    init {
        floatField = 4.2.toFloat()

        privateField = privateField * 2 // stop compiler warning
    }

    fun method1(args: Array<String?>?): Array<String?>? {
        return args
    }

    @kotlin.Throws(Exception::class)
    fun failmethod2(): Int {
        val x = 45
        if (x == 45) throw Exception("death in failmethod2()")
        return 43
    }

    companion object {
        //
        // Class for testing the FFI code with.
        //
        var staticField: Int = 123
        fun staticMethod1(count: Int): Array<String?> {
            val retval = arrayOfNulls<String>(count)
            for (i in 0..<count) {
                retval[i] = Integer.toString(i)
            }
            return retval
        }

        fun staticMethod2(count: Int): Array<Int?> {
            val retval = arrayOfNulls<Int>(count)
            for (i in 0..<count) {
                retval[i] = Integer.valueOf(i)
            }
            return retval
        }

        @kotlin.Throws(Exception::class)
        fun failmethod1(): Int {
            val x = 45
            if (x == 45) throw Exception("death in failmethod1()")
            return 43
        }
    }
}
